packages/backend-plugin-api/src/services/definitions/AuditorService.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ErrorLike } from '@backstage/errors';
+import type { JsonObject } from '@backstage/types';
+import type { Request } from 'express';
+
+/**
+ * Indicates the event was successful.
+ *
+ * @public
+ */
+export type AuditorEventSuccessStatus = { status: 'succeeded' };
+
+/**
+ * Indicates the event failed and includes details about the encountered errors.
+ *
+ * @public
+ */
+export type AuditorEventFailureStatus<E = ErrorLike> = {
+  status: 'failed';
+  errors: E[];
+};
+
+/**
+ * Indicates the event is unknown.
+ *
+ * @public
+ */
+export type AuditorEventUnknownStatus = { status: 'unknown' };
+
+/**
+ * @public
+ */
+export type AuditorEventStatus =
+  | AuditorEventSuccessStatus
+  | AuditorEventFailureStatus
+  | AuditorEventUnknownStatus;
```

Comment:
Status seems a bit tricky to me, iirc we concluded that we primarily want to log when someone attempts to make a change, at that point we don't know the outcome yet. More discussion on that in the catalog change though

Reply:
The `unknown` status serves as a catch-all for those situations where the specific outcome isn't yet determined or available. It acknowledges the uncertainty and provides a placeholder until a definitive status can be assigned. 😄

Reply:
With the new `createEvent` API we can remove the status types right? That just becomes part of the implementation itself instead

Reply:
`AuditorEventStatus` still needs to be exposed because `.log()` is public.